### PR TITLE
Alfoa/issue1023

### LINKED ISSUE #1023: batchSize=1 for paralell runs with mpi
--------
Issue Description
--------
Putting the batchsize = 1 in the parallel description leads to an execution error ("node" files are not created).
**Input** 
```
    <batchSize>1</batchSize>
    <NumMPI>2</NumMPI>
    <internalParallel>False</internalParallel>
    <mode>
      mpi
      <runQSUB/>
      <memory>6gb</memory>
    </mode>
    <expectedTime>1:00:00</expectedTime>
```
**Error**
```
Exception in thread XXX++1:
Traceback (most recent call last):
  File "/home/XXX/.conda/envs/raven_libraries/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/XXX/.conda/envs/raven_libraries/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/home/XXX/project_TREAT/raven/framework/Runners/SharedMemoryRunner.py", line 136, in <lambda>
    self.thread = InterruptibleThread(target = lambda q, *arg : q.append(self.functionToRun(*arg)),
  File "/home/XXX/project_TREAT/raven/framework/Models/Code.py", line 479, in evaluateSample
    inputFiles = self.createNewInput(myInput, samplerType, **kwargs)
  File "/home/XXX/project_TREAT/raven/framework/Models/Code.py", line 383, in createNewInput
    newInput    = self.code.createNewInput(newInputSet,self.oriInputFiles,samplerType,**copy.deepcopy(kwargs))
  File "/home/XXX/project_TREAT/raven/framework/CodeInterfaces/RAVEN/RAVENInterface.py", line 253, in createNewInput
    raise IOError(self.printTag+' ERROR: The nodefile "'+str(nodeFileToUse)+'" does not exist!')
OSError: RAVEN INTERFACE ERROR: The nodefile "/home/XXX/./node_0" does not exist!

```
while the following completes without error
```
    <batchSize>2</batchSize>
    <NumMPI>2</NumMPI>
    <internalParallel>False</internalParallel>
    <mode>
      mpi
      <runQSUB/>
      <memory>6gb</memory>
    </mode>
    <expectedTime>1:00:00</expectedTime>
```

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

Closes #1023

##### What are the significant changes in functionality due to this change request?

added a test. The issue is not valid anymore.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: tests/cluster_tests/test_mpiqsub_1_proc.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    <Triangular name="CladFailureDist">
+      <apex>1</apex>
+      <min>-0.1</min>
+      <max>3</max>
+      <!--
+        <lowerBound>0</lowerBound>
+         This is not yet supported
+        <upperBound>2</upperBound>
+        <adjustement>up</adjustement>
+      -->
```
Comment (wangcj05) [inline tests/cluster_tests/test_mpiqsub_1_proc.xml:64]: Do you want to keep the commented lines?

File: tests/cluster_tests/test_mpiqsub_1_proc.xml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    </Triangular>
+  </Distributions>
```
Comment (wangcj05) [inline tests/cluster_tests/test_mpiqsub_1_proc.xml:66]: Distributions are not used?

Comment (alfoa) [reply tests/cluster_tests/test_mpiqsub_1_proc.xml:66]: removed

**Comment by alfoa** [comment]
@wangcj05 it should be okay now

**Comment by wangcj05** [comment]
@alfoa  Does this PR trigger the regression tests? I could not see the test result.

**Comment by wangcj05** [comment]
@aalfonsi @alfoa There is a problem with the windows machine. Please check it. When the tests are green, this PR can be merged.

**Comment by alfoa** [comment]
@wangcj05 I restarted it

**Comment by wangcj05** [comment]
test failed on Windows. @alfoa Could you talk a look at the windows test? You probably need to increase the tolerance?

**Comment by alfoa** [comment]
@wangcj05 @PaulTalbot-INL approved?

**Comment by PaulTalbot-INL** [comment]
Tests pass, which was the only checklist item remaining since the last code change. Merging.

**Pull request merged**